### PR TITLE
Add support for Forwarded and X-Forwarded headers

### PR DESCRIPTION
**Pull request by dain**
Support Forwarded and X-Forwarded headers in proxied environments

### CODE REVIEW

File: http-server/src/main/java/io/airlift/http/server/HttpServerConfig.java [hunk 1]
```diff
@@ -351,6 +353,19 @@ public HttpServerConfig setLogMaxFileSize(DataSize logMaxFileSize)
         return this;
     }
 
+    public boolean isForwardedEnabled()
+    {
+        return forwardedEnabled;
+    }
+
+    @Config("http-server.forwarded.enabled")
```
Comment (findepi) [inline http-server/src/main/java/io/airlift/http/server/HttpServerConfig.java:361]: "process-forwarded", "handle-forwaded"?

File: http-server/src/test/java/io/airlift/http/server/ForwardedServlet.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2010 Proofpoint, Inc.
```
Comment (findepi) [inline http-server/src/test/java/io/airlift/http/server/ForwardedServlet.java:2]: :)

File: http-server/src/main/java/io/airlift/http/server/HttpServerConfig.java (hunk 1 above)
Comment (electrum) [reply http-server/src/main/java/io/airlift/http/server/HttpServerConfig.java:361]: I like `http-server.process-forwarded`

File: http-server/src/main/java/io/airlift/http/server/HttpServerConfig.java [hunk 3]
```diff
@@ -351,6 +353,19 @@ public HttpServerConfig setLogMaxFileSize(DataSize logMaxFileSize)
...
+    @ConfigDescription("Support Forwarded and X-Forwarded headers in proxied environments")
```
Comment (electrum) [inline http-server/src/main/java/io/airlift/http/server/HttpServerConfig.java:362]: Support -> Process

File: http-server/src/test/java/io/airlift/http/server/ForwardedServlet.java (hunk 2 above)
Comment (electrum) [reply http-server/src/test/java/io/airlift/http/server/ForwardedServlet.java:2]: Yep, I think we can remove this

**Pull request merged**